config/routes.rb

```diff
@@ -23,7 +24,7 @@
   resources :projects, only: [:show] do
     resources :activities, only: [] do
       collection do
-        get :poll, constraints: { format: /js/ }
+        get :poll, defaults: { format: :js }
```

Comment:
this doesn't seem related? Why is it needed?

Reply:
When I was checking with the browser console what was going on, I noticed that polling requests when the session expired tried to be redirected.
That is because rails was processing them as */* and in the failure app they were redirected as html format.
With this the poll requests is processed as js in the failure app the get a 404, because that is what we give to any html format.
Before, when a poll was received for and the session expired:
<img width="1272" alt="captura de pantalla 2019-02-21 a las 21 56 05" src="https://user-images.githubusercontent.com/85766/53204697-30629380-3624-11e9-9016-b67b990728fb.png">
```
Processing by SessionsController#failure as */*
  Parameters: {"last_poll"=>"1550786143", "project_id"=>"1"}
Redirected to http://localhost:3000/login
Completed 302 Found in 1ms (ActiveRecord: 0.0ms)
```
And now it is:
<img width="1272" alt="captura de pantalla 2019-02-21 a las 21 57 24" src="https://user-images.githubusercontent.com/85766/53204726-41130980-3624-11e9-8a92-b4cc4f385893.png">
```
Processing by SessionsController#failure as JS
  Parameters: {"last_poll"=>"1550786228", "project_id"=>"1"}
Completed 404 Not Found in 1ms (ActiveRecord: 0.0ms)
```

Reply:
^ wait, scratch this please. I am seeing a different result when I try this on `master` branch now. The `poll` request gets the html login form now but I see no redirection.

Reply:
Ok, I figured it out:

Now in `master` branch, when we get an expired `poll` request, it is processed by the current failure app `SessionController.new`, so it tries to render the login form with a `Session Expired` message:
<img width="1248" alt="captura de pantalla 2019-02-25 a las 10 39 30" src="https://user-images.githubusercontent.com/85766/53332012-32f60f00-38ea-11e9-8d78-f95a142a20f9.png">
(it is a js request and nobody sees that)

In this PR branch, we are introducing a new failure app: `SessionsController.failure`, that redirects expired html requests to login forms. That's why when we got an expired `poll` request, we were seeing a redirect:
<img width="1272" alt="captura de pantalla 2019-02-21 a las 21 56 05" src="https://user-images.githubusercontent.com/85766/53332264-c92a3500-38ea-11e9-854d-aca06a52db24.png">
The `poll` request format was not `js`, was `*/*`, so in the failure app it was handled as html.

With this change: `get :poll, defaults: { format: :js }` we are making the `poll` requests be processed as `js` requests:
<img width="1272" alt="captura de pantalla 2019-02-21 a las 21 57 24" src="https://user-images.githubusercontent.com/85766/53332698-e27fb100-38eb-11e9-8c58-514f9fada411.png">
Then when we get an expired `js` request we are just returning `head :unknown`.

Reply:
I think the spirit of the original line was: only allow JS calls to this endpoint.

The problem was the client-side poller wasn't hitting `/poll.js` but `/poll`.

With this change, we're making any calls hitting `/poll` to be considered JS calls. Which works, however, is that what we want? I think that the other way to fix it (making the JS poller actually hit the JS endpoing) is preferable / clearer / more consistent with other JS calls (tree navigation for instance, console logger)?

In any case, we can move this to a separate PR, as the rest of the Warden-related code is ready to merge.

Reply:
I understand what you say, and I think it was my other option to do it. I think it can be done like this: https://github.com/dradis/dradis-ce/pull/439/commits/3a4e524b4360b66b5f4030268e5186a2cc3212eb